store/v2/multi/migration.go

```diff
@@ -0,0 +1,68 @@
+package multi
+
+import (
+	dbm "github.com/cosmos/cosmos-sdk/db"
+	"github.com/cosmos/cosmos-sdk/store/iavl"
+	"github.com/cosmos/cosmos-sdk/store/mem"
+	v1Store "github.com/cosmos/cosmos-sdk/store/rootmulti"
+	"github.com/cosmos/cosmos-sdk/store/transient"
+	"github.com/cosmos/cosmos-sdk/store/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// MigrateV2 will migrate the state from iavl to smt
+func MigrateV2(rootMultiStore *v1Store.Store, store2db dbm.DBConnection, storeConfig StoreConfig) (*Store, error) {
+	type namedStore struct {
+		*iavl.Store
+		name string
+	}
+	var stores []namedStore
+	for _, storeKey := range rootMultiStore.GetStoreNames() {
+		keyName := storeKey.Name()
+		switch store := rootMultiStore.GetStoreByName(keyName).(type) {
+		case *iavl.Store:
+			err := storeConfig.RegisterSubstore(keyName, types.StoreTypePersistent)
+			if err != nil {
+				return nil, err
+			}
+			stores = append(stores, namedStore{name: keyName, Store: store})
+		case *transient.Store, *mem.Store:
+			continue
+		default:
+			return nil, sdkerrors.Wrapf(sdkerrors.ErrLogic, "don't know how to migrate store %q of type %T", keyName, store)
+		}
+	}
+
+	// creating the new store of smt tree
+	rootStore, err := NewStore(store2db, storeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// if version is 0 there is no state data to commit
+	if rootMultiStore.LastCommitID().Version == 0 {
+		return rootStore, nil
+	}
+
+	// iterate through the rootmulti stores and save the key/values into smt tree
+	for _, store := range stores {
+		subStore, err := rootStore.getSubstore(store.name)
```

Comment:
shouldn't we make sure to that we use the same store key when doing migration?

Reply:
we are registering the `storeKeys` into v2Store from v1Store 
https://github.com/cosmos/cosmos-sdk/pull/10962/files#diff-ad3a972d3e8f738c0ee269b134134e2beee4e5f0ee9849b0c71eb7cc9d669ce1R24